block.go

```diff
@@ -539,6 +539,13 @@ func (pb *Block) Snapshot(dir string) error {
 	return nil
 }
 
+// Returns true if the block overlaps [mint, maxt].
```

Comment:
Bit confused. You have 2 method name `OverlapsClosedInterval` and comment `Returns true if the block overlaps [mint, maxt].`  but with different logic.

The inner comments and logic make sense though. I think we could leave it as methods and name it just `isOverlapping` (or `...halfOpen..` for block and `...closed...` for chunk) and move inner comments to method comment.

Reply:
These methods are good place to explain the difference in time ranges between block and chunks maybe?

Reply:
And you're right, I should expand on the difference between block and chunk intervals in those method descriptions.

Reply:
Wait, but one is Closed, second is Half-Closed interval, right?

Reply:
Ok, this "closed" is related to arguments... hm this is super confusing (for me), I would just leave `OverlapsInterval` for both chunk and block in this case.

Reply:
or even `Overlaps` .. basically we want to know if `mint` and `maxt` overlaps with block or chunk. And it gives us right answer (e.g: if `mint` is touching `block.maxt` is does NOT overlap)

Reply:
I'm not 100% convinced it's clearer to just call it `Overlaps`. I wanted to highlight the fact the `mint` and `maxt` arguments refer to a closed interval to avoid someone using it to compare a block with another block, in which case `[mint, maxt)` would be a half-open interval, whereas the method expects it to be closed.

That being said, if it's the only thing keeping this PR from being merged, I'm happy to make the change.

Reply:
1. We need someone from maintainer to approve this PR anyway, so maybe someone can put his view on this as well, I don't want to block this PR either (: 
2. If I do `block1.Overlaps(block2.min, block2.max)` is there anything confusing by the fact that it will check overlaps using `[a, b)` because blocks are like this? I think having something like `block1.OverlapsClosedInterval(block2.min, block2.max)` would be confusing for future readers of this code, because it is saying `closed` but actually it is half open overlap logic. But I might be wrong here!

Reply:
But `block1.OverlapsClosedInterval(block2.min, block2.max)` would return the wrong result, because `[block2.min, block2.max)` isn't a closed interval. You would get `block1.min <= block2.max && block2.min < block1.max` instead of the correct value for two `[a, b)` intervals, which is `block1.min < block2.max && block2.min < block1.max`.

That's why I wanted the method name to explicitly state that it requires a closed interval as its argument; it doesn't work on other types of intervals.

Reply:
Got you, agree (: